### ISSUE TITLE
Problem with Relative URL

### ISSUE THREAD

**Issue by Tango43** [CIRCL/AIL-framework#236]
Hello,
We have a question concerning relative Urls redirections for ail-framework. We have an apache proxy working as a reverse proxy which redirects based on Url extension meaning that for AIL we type “ourdomain”/ail to connect to it and the reverse proxy redirects to the AIL server. However, this possess problem as the AIL server loads JS and CSS at hardcoded locations like “/static/” these js and css are therefore not found on the reverse proxy, because there is no way for the reverse proxy to know it should forward it to the AIL server.

On many applications there is an external Url setting which allows the location of the js and css scripts to be defined. Or a base path setting which forces the usage of a base path in front of the Urls and therefor allows usage of these relative urls.

We wanted to avoid having to use a separate Domain name for our different applications. And would preferably also prefer to avoid having to rewrite all the path for the website in a static manner.

Would you either know a solution? Or In case there is none be up for a config setting which forces a base path for relative urls if set? For example in this case it would force “ourdomain”/ail/static instead of just redirecting to “ourdomain”/static/ which off-course fails. 
For us a good example of this implementation is the way this is done in Kibana (ELK stack), where a simple /elk base path allows us to force paths of the requests to “ourdomain”/elk/js and “ourdomain”/elk/css.

**Comment by Tango43** [comment]
Yes this would fix the issues we are encountering. As the requests would be able to pass the proxy.

**Comment by mokaddem** [comment]
Perfect!
We will ping you when the feature will be live.

**Comment by Terrtia** [comment]
Hey @Tango43,

I just added this option (5eadc2755f79e30eda5fade54e7d96bd19d228ac) . You can change it in the bin/config.cfg file (It's in the Flask section). 

Can you please try it and tell me if this fix your issue ?

**Comment by Tango43** [comment]
Hello, I confirm the webpages are hosted correctly. with the extension i specify in base url.
However some of the scripts are still being retreived without the base url option encforced for example i have set base url to /ail in the config but in my browser I can see 404 answers for ip/_stuff, when it should be ip/ail/_stuff.
This means that while the website works, none of javascripts work.

**Comment by Terrtia** [comment]
This should be fix with ff0affc679410a5792cd65af6c2755cb532610d8, I removed some hard-coded links.

Can you please try it ?

Tell me if you found any 404 errors or hard-coded url

**Comment by Terrtia** [comment]
Is the term added ?

Can you please check if this link give you a json: http://localhost:7000/terms_management_query/?term=str_rot13&section=followTerm

**Comment by Tango43** [comment]
I do get a json reponse which this query but it says: {0,0,0,No date recorded.}
It definately seems like the terms do not get stored properly.
It is behind a apache reverse-proxy i tried also connecting when port forwarded in local and i have the same problem. Everything works aparts from adding a term to term management.

**Comment by be-mot** [comment]
I have a similar issue that I can't seem to get resolved.
I have a nginx reverse proxy with config
`        location /ail {
                proxy_pass http://xxx.xxx.xxx.xxx:7000;
                proxy_set_header Host $host;
                proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
                proxy_set_header X-Scheme $scheme;
                proxy_redirect     off;
        }`

In the AIL-config I have
`baseUrl = /ail`

With this everything works except when in the "terms management" page I press the litte "i" icon to see the list of matched pastes:
- The modal opens but remains in "Loading"
- I don't see any HTTP call-out happening via the F12 network view

This with an AIL-version installed +- 1 month ago.

I tried to figure out the issue but haven't managed so far :-(
